splice_machine/src/main/java/com/splicemachine/derby/jdbc/SpliceTransactionResourceImpl.java

```diff
@@ -124,8 +133,11 @@ public boolean marshallTransaction(TxnView txn, ManagedCache<String, Optional<St
 
 
     public void close(){
-        csf.resetCurrentContextManager(cm);
-        csf.removeContextManager(cm);
+        if (prepared) {
```

Comment:
_prepared_ is not the right condition here. Should be _updated_.
If _close()_ is called from the catch block above (l.129) ContextManager is not reset/removed as _prepared = false_ => memory leak.

Reply:
Very good point, thanks!

Reply:
It's rather cosmetic now but instead of having _prepared_ you could just check for _cm == null_ (and enforce that in _close()_).